chronos/profiler/metrics.nim

```diff
@@ -119,14 +146,11 @@ proc futureCompleted(self: var ProfilerMetrics, event: Event): void =
     self.partials.del(event.futureId)
 
 proc processEvent*(self: var ProfilerMetrics, event: Event): void =
-  case event.newState:
-  of Pending: self.futureCreated(event)
-  of Running: self.futureRunning(event)
-  of Paused: self.futurePaused(event)
-  # Completion, failure and cancellation are currently handled the same way.
-  of Completed: self.futureCompleted(event)
-  of Failed: self.futureCompleted(event)
-  of Cancelled: self.futureCompleted(event)
+  case event.event:
```

Comment:
I wouldn't rename the states. I'd keep them in line with the state names that Chronos uses, otherwise we're multiplying terminology. And I think it also communicates intent more clearly - we're extending the future state machine with 2 extra events (running/paused), might even get the Chronos folks to consider putting those in their FutureState enum.

I would also not remove the distinction between the finish states cause we may want to count them.

Reply:
> I wouldn't rename the states. I'd keep them in line with the state names that Chronos uses, otherwise we're multiplying terminology. And I think it also communicates intent more clearly - we're extending the future state machine with 2 extra states (running/paused).

Yah I started viewing them as orthogonal concepts. Chronos is currently written where FutureState is about the result of the Future. Meanwhile we're more concerned with the execution states. Adding the 2 running/paused states subtly overloads the intent of the original Chrono's FutureState's. I think that's more confusing as you need to re-envision FutureState as a slightly different state machine. 

If we could add running/paused to the FutureState itself I think that'd change implicit state machine and but would require modifying other parts of Chronos. 

Having two distinct state machines helps clarify what each API is tracking. It seems sorta weird to me to have an api where we'd expose an `ExtendedFutureState` but that you need to know that only running/paused are valid options, but otherwise you need to use the existing FutureState value(s). Or as you did you need to expose two different callbacks which encode that slightly different intent. 

Initially I tried three states like `ExecutionStates = enum Default, Running, Paused` and then using that to make the `ExtendedFutureState` internally.

I don't know if that makes sense. I'm not super adamant on it, but I think it'd be nice to have one callback.

> I would also not remove the distinction between the finish states cause we may want to count them. 

We still get the FutureState which gives us that info.